hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -1094,6 +1094,11 @@ object HoodieSparkSqlWriter {
     if (mergedParams.contains(PRECOMBINE_FIELD.key())) {
       mergedParams.put(HoodiePayloadProps.PAYLOAD_ORDERING_FIELD_PROP_KEY, mergedParams(PRECOMBINE_FIELD.key()))
     }
+    if (mergedParams.get(OPERATION.key()).get == INSERT_OPERATION_OPT_VAL && mergedParams.contains(DataSourceWriteOptions.INSERT_DUP_POLICY.key())
+      && mergedParams.get(DataSourceWriteOptions.INSERT_DUP_POLICY.key()).get != FAIL_INSERT_DUP_POLICY) {
+      // enable merge allow duplicates when operation type is insert
+      mergedParams.put(HoodieWriteConfig.MERGE_ALLOW_DUPLICATE_ON_INSERTS_ENABLE.key(), "true")
```

Comment:
I feel by default, we should never dedup for INSERT operation. That keeps the behavior in line with regular RDBMS.

Reply:
There is no much meaningness to keep backwards compatible if the behavior itself is not correct from user's intuition,
because most of the users that use `INSERT` operation does not need deduplication and they do not want to specify a record key either.

Reply:
this is not de-dup. this is actually achieving what you are claiming Danny. 
i.e. 
if you ingest RK1, val1 in commit and RK1, val2 in commit2 with insert operation type, snapshot will return both values only when you set "MERGE_ALLOW_DUPLICATE_ON_INSERTS_ENABLE.key" = true.

Reply:
Yeah, kind of obscure from the first sight, why not just put the default value `hoodie.merge.allow.duplicate.on.inserts` as true.

Reply:
yeah, it's better we can keep the strategy in line.

Reply:
nope. we are streamlining across spark-ds and spark-sql. 
if operation type is insert, we do enable hoodie.merge.allow.duplicate.on.inserts if user does not explicitly set it.

Reply:
> we do enable hoodie.merge.allow.duplicate.on.inserts

Do you mean disable?

Reply:
Then it's alright. Somehow while reviewing this change i thought it's in `ProvidesHoodieConfig`. Only now I realised it's in `HoodieSparkSqlWriter` so should be fine.